File: ClassPrevious/class/01/Hw2/Hw2/OGLLight.java
//////////////////////////////////////////////////////////////////////////
//
//  This code is for instructional purposes only. It was generated for
//  use in a graduate level course to show certain aspects of data
//  storage algorithms. It has problems and should not be used
//  outside the class environment.
//
//  Author:  Thomas D. Citriniti     citrit@rpi.edu
//  Class:   Advanced Computer Graphics and Data visualization
//           Rensselaer Polytechnic Institute
//  Date:    January 1998
//
//////////////////////////////////////////////////////////////////////////

import jogl.*;

/**
  * Light class to handle light positions.
  */
public class OGLLight extends Light
{

	/** Constructor */
	public OGLLight(int num) { super(num); }

	/** Called by the Renderer to initialize the light */
	public void render(Renderer aren)
	{
		JoglCanvas gl = ((OGLRenderer)aren).gl;
		int lightnum = 0;

		float lightPos[] = {from.x, from.y, from.z, 0.0F};
		float lightDir[] = {to.x, to.y, to.z, 0.0F};

		gl.use();
		if (on == true) {
		    switch (number) {
				case 0: lightnum = GL.LIGHT0; break;
				case 1: lightnum = GL.LIGHT1; break;
				case 2: lightnum = GL.LIGHT2; break;
				case 3: lightnum = GL.LIGHT3; break;
				case 4: lightnum = GL.LIGHT4; break;
				case 5: lightnum = GL.LIGHT5; break;
				case 6: lightnum = GL.LIGHT6; break;
				case 7: lightnum = GL.LIGHT7; break;
			}
			gl.enable(GL.LIGHTING);
			gl.enable(lightnum);
			gl.light(lightnum, GL.POSITION, lightPos);
			gl.light(lightnum, GL.SPOT_DIRECTION, lightDir);
			if (color != null) {
				//gl.light(lightnum, GL.AMBIENT, color.ambient);
				gl.light(lightnum, GL.DIFFUSE, color.diffuse);
				gl.light(lightnum, GL.SPECULAR, color.ambient);
			}
		}
		else {
			gl.disable(GL.LIGHTING);
		}
	}
}

File: WebGL/OGLRen/OGLRen/OGLLight.js
//////////////////////////////////////////////////////////////////////////
//
//  This code is for instructional purposes only. It was generated for
//  use in a college course to show certain aspects of data
//  storage algorithms. It has problems and should not be used
//  outside the class environment.
//
//  Author:  Thomas D. Citriniti     citrit@rpi.edu
//  Class:   Computer Graphics 
//           Rensselaer Polytechnic Institute
//  Date:    Sept 2015
//
//////////////////////////////////////////////////////////////////////////

var OGLRen = OGLRen || {};

/**
  * WebGL Light class
  */
OGLRen.OGLLight = function(lnum) {
    /** Called by the Renderer to initialize the light */
	this.render = function( aren)
	{
		var gl = aren.getGL();
		var lightnum = 0;
        var from = new OGLRen.PointType(10, 10, 10);
		var to = new OGLRen.PointType();
		color = new OGLRen.Material();
		var lightPos = [from.x, from.y, from.z, 0.0];
		var lightDir = [to.x, to.y, to.z, 0.0];

		if (this.on == true) {
		    switch (this.number) {
				case 0: lightnum = gl.LIGHT0; break;
				case 1: lightnum = gl.LIGHT1; break;
				case 2: lightnum = gl.LIGHT2; break;
				case 3: lightnum = gl.LIGHT3; break;
				case 4: lightnum = gl.LIGHT4; break;
				case 5: lightnum = gl.LIGHT5; break;
				case 6: lightnum = gl.LIGHT6; break;
				case 7: lightnum = gl.LIGHT7; break;
			}
			//gl.enable(gl.LIGHTING);
			//gl.enable(lightnum);
			//gl.light(lightnum, gl.POSITION, lightPos);
			//gl.light(lightnum, gl.SPOT_DIRECTION, lightDir);
			//if (color != null) {
				//gl.light(lightnum, gl.AMBIENT, this.getColor().ambient);
				//gl.light(lightnum, gl.DIFFUSE, this.getColor().diffuse);
				//gl.light(lightnum, gl.SPECULAR, this.getColor().ambient);
			//}
		}
		else {
			//gl.disable(gl.LIGHTING);
		}
	}
}
OGLRen.OGLLight.inheritsFrom(OGLRen.Light);

